app/ts/ManualInput/CellValidityService/CellValidityService.test.ts

```diff
@@ -367,4 +372,273 @@ describe('CellValidityService', () => {
+      expect(result).toBeTruthy();
+    });
+
+    it('should return true if effect is not of type "value"', () => {
+      const effect = {type: 'valueCI'} as Effect;
+      const result = isValue(effect);
+      expect(result).toBeFalsy();
+    });
+  });
+
+  describe('isValueCI', () => {
+    it('should return true if effect is of type "valueCI"', () => {
+      const effect = {type: 'valueCI'} as Effect;
+      const result = isValueCI(effect);
+      expect(result).toBeTruthy();
+    });
+
+    it('should return true if effect is not of type "valueCI"', () => {
+      const effect = {type: 'value'} as Effect;
+      const result = isValueCI(effect);
+      expect(result).toBeFalsy();
+    });
+  });
+
+  describe('isRange', () => {
+    it('should return true if effect is of type "range"', () => {
+      const effect = {type: 'range'} as Effect;
+      const result = isRange(effect);
+      expect(result).toBeTruthy();
+    });
+
+    it('should return true if effect is not of type "range"', () => {
+      const effect = {type: 'value'} as Effect;
+      const result = isRange(effect);
+      expect(result).toBeFalsy();
+    });
+  });
+
+  describe('isNormalDistribution', () => {
+    it('should return true if distribution is of type "range"', () => {
+      const distribution = {type: 'normal'} as Distribution;
+      const result = isNormalDistribution(distribution);
+      expect(result).toBeTruthy();
+    });
+
+    it('should return true if distribution is not of type "range"', () => {
+      const distribution = {type: 'beta'} as Distribution;
+      const result = isNormalDistribution(distribution);
+      expect(result).toBeFalsy();
+    });
+  });
+
+  describe('getOutOfBoundsError', () => {
+    it('should return an error if a value effect is out of bounds', () => {
+      const datasourceId = 'ds1Id';
+      const effects: Record<string, Record<string, Effect>> = {
+        ds1Id: {
+          alt1Id: {
+            criterionId: 'crit1Id',
+            dataSourceId: 'ds1Id',
+            alternativeId: 'alt1Id',
+            type: 'value',
+            value: 10
+          }
+        },
+        ds2Id: {
+          alt1Id: {
+            criterionId: 'crit1Id',
+            dataSourceId: 'ds2Id',
+            alternativeId: 'alt1Id',
+            type: 'value',
+            value: 0.5
+          }
+        }
+      };
+      const distributions: Record<string, Record<string, Distribution>> = {};
+      const inputUpperBound = 1;
+      const result = getOutOfBoundsError(
+        datasourceId,
+        effects,
+        distributions,
+        inputUpperBound
+      );
+      const expectedResult = 'Some cell values are out of bounds [0, 1]';
+      expect(result).toEqual(expectedResult);
+    });
+
+    it('should return an error if a valueCI effect is out of bounds', () => {
+      const datasourceId = 'ds1Id';
+      const effects: Record<string, Record<string, Effect>> = {
+        ds1Id: {
```

Comment:
If you're always using the same effects structure, use function to construct the base and modify it for test requirements